tests/mocks/lib/api.c

```diff
@@ -26,7 +26,7 @@ struct mocked_api {
     u64      stop2;
     union rc rc2;
 
-    atomic64_t calls HSE_L1D_ALIGNED;
+    atomic_ulong calls HSE_L1D_ALIGNED;
```

Comment:
Is atomic_ulong guaranteed to be 64-bit on all platforms like atomic64_t seemed to be?

Reply:
Right, so atomic64_t was originally just a long, but I recently changed it to atomic_long.
On a 32-bit platform ulong and atomic_ulong are probably only 32 bits.  But we can't run on a 32-bit platform, so no worry there.  I'm not aware of any 128-bit platforms were ulong would be 128 bits, but we can cross that bridge if that ever happens.  There are some alleged 64-bit atomic types (e.g., least and fast types), but it appeared that the "fast32" type is actually 8 bytes long, so I decided to avoid that...  It's plausible we could use atomic_uint_fast64_t instead of atomic_ulong, but after this it would be a simple perl replace operation.  The thing I have been focusing on is getting the "signedness" correct.  Open to suggestions, I need to do some more research on these "fast" types to understand the semantics..

Reply:
I am not familiar with the the various fast/least types either, but I was mainly referring to using `typedef _Atomic(uint64_t) atomic_u64`.

Reply:
Right.  No, I hadn't tried that, I'm little suspicous of creating our own atomic types, would need to investigate it first...

Reply:
both the atomic_uint_fast32_t and atomic_uint_fast64_t types are defined at "lon gunsigned int" by gcc my machines.  YMMV, but they have to be at least 32/64 bits wide, but can be larger according to the spec.  It's left to the implementation...

Reply:
My usual source of reading: https://en.cppreference.com/w/c/language/atomic

Reply:
Yeah, that's a good page, but the spec has all the gory details.   C99 had a rationale tome that accompanied the spec which was three or four times as large as the spec.  But I haven't been able to find a similar rationale "paper" for the latest C spec...